src/sentry/snuba/discover.py

```diff
@@ -735,6 +735,129 @@ def get_facets(query, params, limit=10, referrer=None):
     return results
 
 
+def get_performance_facets(
+    query, params, aggregate_column="duration", aggregate_function="avg", limit=20, referrer=None
+):
+    """
+    High-level API for getting 'facet map' results for performance data
+
+    Performance facets are high frequency tags and the aggregate duration of
+    their most frequent values
+
+    query (str) Filter query string to create conditions from.
+    params (Dict[str, str]) Filtering parameters with start, end, project_id, environment
+    limit (int) The number of records to fetch.
+    referrer (str|None) A referrer string to help locate the origin of this query.
+
+    Returns Sequence[FacetResult]
+    """
+    with sentry_sdk.start_span(
+        op="discover.discover", description="facets.filter_transform"
+    ) as span:
+        span.set_data("query", query)
+        snuba_filter = get_filter(query, params)
+
+        # Resolve the public aliases into the discover dataset names.
+        snuba_filter, translated_columns = resolve_discover_aliases(snuba_filter)
+
+    # Exclude tracing tags as they are noisy and generally not helpful.
+    # TODO(markus): Tracing tags are no longer written but may still reside in DB.
+    excluded_tags = ["tags_key", "NOT IN", ["trace", "trace.ctx", "trace.span", "project"]]
+
+    # Sampling keys for multi-project results as we don't need accuracy
+    # with that much data.
+    sample = len(snuba_filter.filter_keys["project_id"]) > 2
+
+    with sentry_sdk.start_span(op="discover.discover", description="facets.frequent_tags"):
+        # Get the tag keys with the highest deviation
+        key_names = raw_query(
+            aggregations=[["stddevSamp", aggregate_column, "stddev"]],
+            start=snuba_filter.start,
+            end=snuba_filter.end,
+            conditions=snuba_filter.conditions,
+            filter_keys=snuba_filter.filter_keys,
+            orderby=["-stddev", "tags_key"],
+            groupby="tags_key",
+            # TODO(Kevan): Check using having vs where before mainlining
+            having=[excluded_tags],
+            dataset=Dataset.Discover,
+            limit=limit,
+            referrer=referrer,
+            turbo=sample,
+        )
+        top_tags = [r["tags_key"] for r in key_names["data"]]
+        if not top_tags:
+            return []
+
+    results = []
+
+    sampling_enabled = True
+    sample_rate = 0.1 if sampling_enabled else None
+
+    max_aggregate_tags = 20
```

Comment:
Discussed offline, but lets just remove `individual_tags` from this entirely, and introduce it back if there is a performance issue instead